OpenRA.Mods.D2k/Traits/AttackSwallow.cs

```diff
@@ -67,5 +69,19 @@ public override void DoAttack(Actor self, Target target, IEnumerable<Armament> a
 			self.CancelActivity();
 			self.QueueActivity(new SwallowActor(self, target, a, facing));
 		}
+
+		public override Activity GetAttackActivity(Actor self, Target newTarget, bool allowMove, bool forceAttack)
+		{
+			return new SwallowTarget(self, newTarget, allowMove, forceAttack, Info.FacingTolerance);
+		}
+
+		public class SwallowTarget : Attack
+		{
+			public SwallowTarget(Actor self, Target target, bool allowMovement, bool forceAttack, int facingTolerance)
+				: base(self, target, allowMovement, forceAttack, facingTolerance) { }
+
+			// Worms ignore visibiity, so don't need to recalculate targets
+			protected override bool IgnoresVisibility { get { return false; } }
```

Comment:
This should return `true`!

Reply:
Ah, good catch.  This is deleted in the next commit, so I didn't bother to test it.